lib/diagnostic.rb

```diff
@@ -43,6 +47,14 @@ def transform(words)
   # replacing the trailing newline character with a
   # space and return the result.
   def read_file(filename)
-    # Your code goes here
+    lines = File.open(ARGV[0]) do |file|
+      file.map(&:itself)
+    end
+    puts lines
   end
 end
+
+# Jason Liang's comments:
+# I copied this directly from the training yesterday
+# This is subject quite unknown to me
+# can you clarify what ARGV[0] means?
```

Comment:
ARGV[0] is a reference to the first argument to the method (similar to `arguments[0]` in JavaScript). No need to use it here, since we have a named first argument.